src/hotspot/share/opto/parseHelper.cpp

```diff
@@ -190,16 +189,12 @@ void Parse::array_store_check() {
       //
       // See issue JDK-8057622 for details.
 
-    always_see_exact_class = true;
```

Comment:
Why is it ok to remove this?
If this branch is not taken, it used to be `false`, and would lead to something different below...

Reply:
The only use of this is to decide if we need to attach a control input to the `LoadKlass`. As the control input is not needed, this can be removed.

Reply:
Got it, thanks!